src/psyclone/domain/lfric/arg_ordering.py

```diff
@@ -134,6 +159,140 @@ def extend(self, list_var_name, var_accesses=None,
             else:
                 self.append(var, mode=mode, var_accesses=var_accesses)
 
+    def get_integer_symbol(self, name, tag=None):
+        '''This function returns a symbol for an integer reference. If the
+        symbol should not already exist in the symbol table, it will
+        be properly declared and added to the table.
+
+        :param str name: name of the integer variable to declare.
+        :param tag: optional tag of the integer variable to declare.
+        :type tag: Optional[str]
+
+        :returns: the symbol for the variable.
+        :rtype: :py:class:`psyclone.psyir.symbols.Symbol
+
+        '''
+        if not tag:
+            tag = name
```

Comment:
I didn't know this was supposed to be the case. I looked at the tags in the debugger, and saw a mixture of 'special' name, and tag==name. For example from dynamo0p3:
```
        for idx, loop in enumerate(loops):
            root_name = f"loop{idx}_start"
            try:
                lbound = sym_table.lookup_with_tag(root_name)
            except KeyError:
                # TODO #1258 the loop bound symbol should be of
                # precision 'i_def'.
                lbound = sym_table.new_symbol(root_name=root_name,
                                              tag=root_name,
                                              symbol_type=DataSymbol,
                                              datatype=INTEGER_TYPE)
```
But I will see if I can fix/update this - which might be affected by other comments (esp. changing the location of where to declare the symbols).

Reply:
Since we have a mixture (and that might not be correct), I don't think this should be fixed in this PR, so I would suggest to leave this, but depends a bit where the various convenience function are ... if they should go to be in the SymbolTable (or global functions), then we need to support not specifying a tag, and then this will be changed. See discussion below.